pkg/sinks/sink.go

```diff
@@ -246,6 +246,14 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 			forwardOpts = append(forwardOpts, sinkforward.WithCallbackUploader(cbPublisher))
 		}
 
+		// Derive the retryStrategy from the spec and add to the forwarder if it is valid
+		retryStrategy := u.VertexInstance.Vertex.Spec.Sink.GetRetryStrategy()
+
+		if ok := u.VertexInstance.Vertex.Spec.Sink.IsValidSinkRetryStrategy(retryStrategy); ok != nil {
```

Comment:
These functions do not make sense - you already have the sink object, you get the retry strategy from the object, and then call a function of the sink object to tell if the strategy is valid...

Reply:
My bad, oversight.
Have updated to check for validation while fetching retryStrategy